packages/superset-ui-core/src/query/types/QueryFormData.ts

```diff
@@ -133,6 +133,14 @@ export type ExtraFormDataOverride = ExtraFormDataOverrideRegular & ExtraFormData
 
 export type ExtraFormData = ExtraFormDataAppend & ExtraFormDataOverride;
 
+export type DrillDownType = {
+  hierarchy: string[];
+  currentIdx: number;
+  filters: QueryObjectFilterClause[];
+}
+
+export type OwnState = JsonObject & { drilldown: DrillDownType; }
```

Comment:
If we want to enforce type checking on `OwnState`, we probably need to introduce some named properties while moving the custom ones into nested ones. Something like
```ts
{
  drilldown: DrillDownType;
  customState: JsonObject;
}
```
 
In the above example the previous `ownState` would be moved into `customState`, while `drilldown` would be a reserved property for drilldown actions. However, that refactor should probably not be done in this PR, so I propose leaving `ownState` as `JsonObject` for now.